Exceptions/BadConfigException.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Modera\ServerCrudBundle\Exceptions;
+
+/**
+ * This exception can be thrown when an invalid configuration found during executing.
+ *
+ * @author    Alex Plaksin <alex.plaksin@modera.org>
+ * @copyright 2016 Modera Foundation
+ */
+class BadConfigException extends \RuntimeException
+{
+    /**
+     * @var string
+     */
+    protected $serviceType;
+
+    /**
+     * @var array
+     */
+    protected $config;
+
+    /**
+     * @var \Exception
+     */
+    protected $exception;
+
+    /**
+     * @return string
+     */
+    public function getServiceType()
```

Comment:
No need for setter methods, once exception is created it rarely has to be modified, if additional meta-information has become available and it is desired to associate it with the exception, the original one must be wrapped instead.